odl/discr/grid.py

```diff
@@ -802,346 +947,104 @@ def __array__(self, dtype=None):
 
     def __repr__(self):
         """Return ``repr(self)``."""
-        inner_str = ', '.join(array1d_repr(vec) for vec in self.coord_vectors)
-        return '{}({})'.format(self.__class__.__name__, inner_str)
+        if self.is_uniform:
+            constructor = 'uniform_grid'
+            posargs = [list(self.min_pt), list(self.max_pt), self.shape]
+            inner_str = signature_string(posargs, [])
+            return '{}({})'.format(constructor, inner_str)
+        else:
+            constructor = self.__class__.__name__
+            posargs = [array1d_repr(v) for v in self.coord_vectors]
+            inner_str = signature_string(posargs, [], sep=[',\n', ', ', ', '],
+                                         mod=['!s', ''])
+            return '{}(\n{}\n)'.format(constructor, indent_rows(inner_str))
 
     def __str__(self):
```

Comment:
Hm, I thought I had started an attempt at erasing this earlier but hit some resistance. When putting together this PR it tickled again in the code-deleting finger, but I resisted the temptation.
But I'm happy to ditch this function.